src/components/MyKiva/BadgeModalContentJourney.vue

```diff
@@ -179,8 +181,9 @@ const getBadgeStatus = index => {
 
 const handleBadgeClick = (event, index) => {
 	// Prevent analytics being logged when non-completed tier is clicked
-	if (!sortedTiers.value[index]?.completedDate) {
+	if (!sortedTiers.value[index]?.completedDate && getBadgeStatus(index) !== BADGE_LOCKED) {
 		event.stopImmediatePropagation();
+		emit('badge-level-clicked', sortedTiers.value[index]?.level);
```

Comment:
You could emit the tier here instead of the level and not have to find it again in `handleBadgeLevelClicked`